### PR TITLE
feat(testing): add overrideTemplate method

### LINKED ISSUE #10685: [testing] TestBed could offer an `overrideTemplate` method
**I'm submitting a ...**  (check one with "x")

```
[ ] bug report
[X] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
As of rc.5, the new TestBed designed to replace TestComponentBuilder is introduced.
But it looks like, if I'm not mistaken, that we lost the equivalent of `overrideTemplate`, forcing us to use

```
TestBed.overrideComponent(RaceComponent, { set: { template: html }})
```

**Expected/desired behavior**

Maybe we could have something like:

```
TestBed.overrideTemplate(RaceComponent, html)
```
- **Angular version:** 2.0.0-rc.5

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes https://github.com/angular/angular/issues/10685

### CODE REVIEW

**Comment by vicb** [comment]
@juliemr @vikerman any thoughts on this one ?

**Comment by juliemr** [comment]
This looks reasonable to me. Can you add a test to https://github.com/angular/angular/blob/master/modules/%40angular/platform-browser/test/testing_public_spec.ts?

We should also set the templateUrl to null.

**Comment by DzmitryShylovich** [comment]
@juliemr done

File: modules/@angular/platform-browser/test/testing_public_spec.ts [hunk 1]
```diff
@@ -356,6 +356,13 @@ export function main() {
             expect(compFixture.nativeElement).toHaveText('transformed hello');
           });
         });
+
+        describe('template', () => {
+          beforeEach(() => TestBed.overrideTemplate(SomeComponent, 'newText'));
+          it('should work', () => {
```
Comment (vicb) [inline modules/@angular/platform-browser/test/testing_public_spec.ts]: - 'should work' is too vague here, please be more specific.
- there is no test for url = null

Comment (DzmitryShylovich) [inline modules/@angular/platform-browser/test/testing_public_spec.ts]: all other tests have the same msg :)
https://github.com/DzmitryShylovich/angular/blob/026355cab8290be905259d1785d1f6f0de35fa08/modules/%40angular/platform-browser/test/testing_public_spec.ts#L339

Comment (DzmitryShylovich) [inline modules/@angular/platform-browser/test/testing_public_spec.ts]: btw why do we need to set `templateUrl` to null?
`template` is processed first [link](https://github.com/angular/angular/blob/master/modules/%40angular/compiler/src/directive_normalizer.ts#L67) so the value of `templateUrl` doesn't really matter

Comment (vicb) [inline modules/@angular/platform-browser/test/testing_public_spec.ts]: > all other tests have the same msg :)

We need to start fixing somewhere. I should have reviewed those changes :)

Comment (vicb) [inline modules/@angular/platform-browser/test/testing_public_spec.ts]: > btw why do we need to set templateUrl to null?

belt and braces

**Comment by DzmitryShylovich** [comment]
@vicb done

**Comment by geyuqiu** [comment]
What is the purpose of overrideTemplate ?

Could we still write test with fixture.debugElement.query(By...)?

Cheers
Yuqiu

**Pull request merged**